src/psyclone/tests/psyir/transformations/profile_test.py

```diff
@@ -814,3 +816,56 @@ def test_omp_transform():
       CALL profile_psy_data%PostEnd'''
 
     assert correct in code
+
+
+def test_auto_invoke_return_last_stmt(parser):
+    ''' Check that using the auto-invoke profiling option avoids including
+    a return statement within the profiling region if it is the last statement
+    in the routine. '''
+    symbol_table = SymbolTable()
+    arg1 = symbol_table.new_symbol(
+        symbol_type=DataSymbol, datatype=REAL_TYPE)
+    zero = Literal("0.0", REAL_TYPE)
+    assign1 = Assignment.create(Reference(arg1), zero)
+    kschedule = KernelSchedule.create(
+        "work", symbol_table, [assign1, Return()])
+
```

Comment:
I have I don't like tests that assemble their own PSyIR tree like this - I find it very hard to understand, I usually put a simple subroutine in a string and parse it. This way I find the test easier to understand, and it feels to me the test is less dependent on implementation details (e.g. you have dependencies to SymbolTable, REAL_TYPE, Literal, ... - while parsing code would only have the normal parsing interface).

Having said that, this test is fine and does what it is supposed to test, and I don't see a need to change it.

Reply:
I did it this way to be API agnostic (since the profiling routine I'm testing is API agnostic). I admit it's not as simple as just parsing some Fortran though.